hudi-client/hudi-spark-client/src/test/java/org/apache/hudi/client/functional/TestHoodieClientOnCopyOnWriteStorage.java

```diff
@@ -771,14 +783,14 @@ public void testSmallInsertHandlingForUpserts() throws Exception {
     Set<String> keys1 = recordsToRecordKeySet(inserts1);
 
     JavaRDD<HoodieRecord> insertRecordsRDD1 = jsc.parallelize(inserts1, 1);
-    List<WriteStatus> statuses = client.upsert(insertRecordsRDD1, commitTime1).collect();
+    JavaRDD<WriteStatus> rawStatuses = client.upsert(insertRecordsRDD1, commitTime1);
+    JavaRDD<WriteStatus> statuses = jsc.parallelize(rawStatuses.collect(), 1);
+    writeClient.commit(commitTime1, statuses, Option.empty(), COMMIT_ACTION, Collections.emptyMap(), Option.empty());
 
-    assertNoWriteErrors(statuses);
```

Comment:
I see many `assertNoWriteErrors` are deleted, is this as expected?

Reply:
Addressed